fields.go

```diff
@@ -116,3 +100,22 @@ func NamedError(key string, err error) Field {
 	}
 	return zap.NamedError(key, &encoders.ErrorEncoder{Source: err})
 }
+
+// ptrValueTypes lists all acceptable pointer types for creating fields.
+type ptrValueTypes interface {
+	~string | ~int | ~int32 | ~int64 | ~uint | ~uint32 | ~uint64 | ~float32 | ~float64 | ~bool
+}
+
+// Ptr safely dereferences a value for use in a log field, rendering a zero value if the
+// value is nil. For example:
+//
+//   var value *string
+//   log.String("key", log.Ptr(value)) // {"key":"key","value":""}
+//
+func Ptr[T ptrValueTypes](value *T) T {
+	if value == nil {
+		var zero T
+		return zero
+	}
+	return *value
+}
```

Comment:
Nice ! I like this implementation it's very readable and It's clear how nil pointers are handled, I wasn't quite sure how Zap.Any was handling it in that large switch case statement ( [reflection magic maybe](https://sourcegraph.com/github.com/uber-go/zap@6f34060764b5ea1367eecda380ba8a9a0de3f0e6/-/blob/field.go?L50) ? lol )